src/Connection/Resource/Stream.php

```diff
@@ -211,6 +211,18 @@ public function write(string $string): int
             throw new RuntimeException('Unable to write to stream', 1);
         }
 
+        if ($result === 0) {
+            if ($this->eof()) {
+                throw new RuntimeException('Stream closed', 1);
+            }
+
+            if (stream_get_meta_data($this->stream)['timed_out']) {
+                throw new RuntimeException('Stream timed out', 1);
```

Comment:
It doesn't make sense to add this additional exception handling, the exception type is the same as well as error code. We already throws exception with same inputs:

`throw new RuntimeException('Unable to write to stream', 1);`

Reply:
Yes we're unable to write to stream but for different reason. It might be useful for the user IMHO (timeout might indicated insufficient network configuration).

Reply:
But how user will distinguish timeout from another reason? By parsing an exception message?